src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -41,17 +43,31 @@ public void RunApiCompat()
                     IAssemblySymbol leftSymbols = new AssemblySymbolLoader().LoadAssembly(apicompatTuples.assemblyName, leftAssemblyStream);
                     IAssemblySymbol rightSymbols = new AssemblySymbolLoader().LoadAssembly(apicompatTuples.assemblyName, rightAssemblyStream);
 
-                    IEnumerable<CompatDifference> differences = _differ.GetDifferences(leftSymbols, rightSymbols);
+                    _log.LogMessage(MessageImportance.Low, apicompatTuples.header);
 
-                    if (differences.Any())
-                    {
-                        _log.LogError(null, apicompatTuples.compatibilityReason);
-                        _log.LogError(null, apicompatTuples.header);
-                    }
+                    IEnumerable<CompatDifference> differences = _differ.GetDifferences(leftSymbols, rightSymbols);
 
                     foreach (CompatDifference difference in differences)
                     {
-                        _log.LogError(difference.DiagnosticId, difference.Message);
+                        // Only include package version when comparing assets from different packages.
+                        string leftOperand = apicompatTuples.leftAssemblyRelativePath;
+                        string rightOperand = apicompatTuples.rightAssemblyRelativePath;
+                        if (!apicompatTuples.leftAssemblyPackagePath.Equals(apicompatTuples.rightAssemblyPackagePath, System.StringComparison.InvariantCultureIgnoreCase))
+                        {
+                            leftOperand = $"[{Path.GetFileNameWithoutExtension(apicompatTuples.leftAssemblyPackagePath)}]{leftOperand}";
+                            rightOperand = $"[{Path.GetFileNameWithoutExtension(apicompatTuples.rightAssemblyPackagePath)}]{rightOperand}";
+                        }
+
+                        _log.LogError(
```

Comment:
NIT: Why not add an override that takes an IDifference plus the operands, then just call the API that checks if the error should be suppressed taking each individual field and creates the suppression object for you?

Reply:
You will eventually create the Suppression object, whether that be at the SuppressionEngine level or at the Log callsite. I'm fine with any design, but what I like about the current one is that it is very easy to see which are all the different suppression types that we log today by just looking for all the callsites of the Suppression constructor.

Reply:
Yeah, I know we will create the `Suppression` object but it might make sense to just do that in a central location rather than every time I want to log an error I would have to create an object. Either way is fine I was trying to avoid duplication. Also to look at which are the difference suppression types we could just look at the `LogError` references.